module-2/backend/week_four.md

```diff
@@ -10,14 +10,38 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+ A way to maintain state over HTTP request/response cycle.
+
 * What’s the difference between a session and a cookie?
+ A session is similiar to  a cookie except it allows you to have authorization/authentication.
+
 * What’s a flash and when do you want to use flashes?
+ Flash is a piece of functionality you get with rails. It lets you display temporary messages to your users.
+
 * Why do people say “HTTP is stateless”?
+ As a result of being  a RESTful app. It doesn't hold information during communication transactions and after they end.
+
 * What’s authentication? Explain.
+ Verifies a user is a specific user. Allows for relative user privacy across your app and for them to access the info specific to them.
+
 * What’s the difference between authentication and authorization?
+  The former confirms who they say they are, the latter deals with what permissions and access they have once authenticated.
+
 * What’s a before filter?
+  It controls what controller functionality your users/admin have access to you depending on what you specify.
```

Comment:
That's one use, but it's any method we want to run before a controller action using `before_action`.